documentation/specs/src/economics/fee-system.md

```diff
@@ -4,4 +4,36 @@ In order to be accepted by the Namada ledger, transactions must pay fees. Transa
 
 Namada transaction fees can be paid in any fungible token which is a member of a whitelist controlled by Namada governance. Governance also sets minimum fee rates (which can be periodically updated so that they are usually sufficient) which transactions must pay in order to be accepted (but they can always pay more to encourage the proposer to prioritise them). When using the shielded pool, transactions can also unshield tokens in order to pay the required fees.
 
-The token whitelist consists of a list of $(T, GP_{min})$ pairs, where $T$ is a token identifier and $GP_{min}$ is the minimum price per unit gas which must be paid by a transaction paying fees using that asset. This whitelist can be updated with a standard governance proposal. All fees collected are paid directly to the block proposer (incentive-compatible, so that side payments are no more profitable).
\ No newline at end of file
+The token whitelist consists of a list of $(T, GP_{min})$ pairs, where $T$ is a token identifier and $GP_{min}$ is the minimum price per unit gas which must be paid by a transaction paying fees using that asset. This whitelist can be updated with a standard governance proposal. All fees collected are paid directly to the block proposer (incentive-compatible, so that side payments are no more profitable).
+
+todo: fees paid to block proposer??
+
+The transaction whitelist held by governance associates to each
+whitelisted transaction a fee $f_{tx}$ per "transaction unit". The
+definition of a transaction unit may vary per VP; some transactions
+may only ever cost the same amount, while a complex MASP transaction
+with many spends, outputs, and converts counts as one unit for each of
+these. The unit of these is the generic gas unit in which price-per-gas
+for tokens is defined above. Whitelisted fees may be updated by
+governance using a standard proposal.
+
+When a transaction is submitted with $n$ units of data, its
+responsibility is to pay the predictable flat fee $n * f_{tx} *
+GP_{min}$, hereafter $F$, in some whitelisted token $T$. Generically,
+this may be done by transferring the token (which may be a simple
+transparent NAM transfer, a bridged multitoken transfer, or the
+transparent output of a shielded transaction) to a native VP address
+designated for fee burning. It shall be lawful for a block proposer to
```

Comment:
why is this necessary? we don't need to burn any fees - this address can just be the block proposer address directly

Reply:
yeah - for the MASP we have to use a special protocol address I guess - for the other txs we could just fill in the address at the time of transaction execution, but a protocol address is fine too

Reply:
but the block proposer shouldn't need to make an extra transaction, the fees should just be distributed to them in EndBlock

Reply:
I agree with Ray that a native account/VP for handling fee distribution to block producers makes the most sense and is the most user-friendly. To my understanding, coinbase is a transaction that is included by the block producer, so in my head it is as simple as extending this coinbase transaction (or adding an additional one depending on ease) to also include withdrawing funds from the address of the native transaction-fee VP?